### PR TITLE
Deduplicate train_p3 and train_p3_centralized examples

### LINKED ISSUE #88: examples/bucket-brigade: deduplicate train_p3 and train_p3_centralized (extract shared core)
## Background

PR #71 added `examples/games/bucket_brigade/train_p3_centralized.rs` as a near-duplicate of `examples/games/bucket_brigade/train_p3.rs`. The Judge review of PR #71 explicitly flagged this and deferred it to a follow-up:

> `examples/games/bucket_brigade/train_p3_centralized.rs` (327 lines) is largely a duplicate of `train_p3.rs`. Once the parity test is solid, it'd be worth a follow-up to fold the critic-on path into `train_p3.rs` behind a flag rather than maintain two parallel scripts.

Concrete sizes on `main` after PR #71 merged:
- `examples/games/bucket_brigade/train_p3.rs` — 298 LOC
- `examples/games/bucket_brigade/train_p3_centralized.rs` — 327 LOC
- Shared core: ~95% identical (env construction, redundancy-penalty aux loss, arg parsing, checkpoint I/O, training loop). The only structural difference is the trainer constructor (`JointMultiAgentTrainer::new` vs `::new_with_centralized_critic`) and 2-3 additional CLI flags for the critic config.

## Goal

Collapse the two binaries into a single example that toggles centralized-critic via a CLI flag.

## Approach (recommended; refine if cleaner emerges)

1. In `train_p3.rs`, add CLI flags:
   - `--centralized-critic` (bool, default false)
   - `--critic-hidden-dim` (default 64)
   - `--critic-lr` (default 3e-4)
   - `--critic-vf-coef` (default 0.5)
   - `--critic-clip-range-vf` (default 0.0)
2. Branch on `--centralized-critic` at trainer construction: choose `JointMultiAgentTrainer::new` (no critic) or `::new_with_centralized_critic` (with critic). All other code paths remain shared.
3. The `--critic-vf-coef 0` case (per the existing `train_p3_centralized.rs` module docs) recovers the no-critic baseline numerically — verify this still holds with the unified binary.
4. Update the binary's module docs to describe both modes.
5. Delete `examples/games/bucket_brigade/train_p3_centralized.rs`.
6. Update `Cargo.toml` to remove the now-deleted `[[example]] train_p3_centralized` entry, if one exists.
7. Update any docs / READMEs that reference `train_p3_centralized` by name (search `grep -rn train_p3_centralized` before deleting).

## Acceptance criteria

- [ ] Single `examples/games/bucket_brigade/train_p3.rs` supports both no-critic and centralized-critic modes via a `--centralized-critic` flag.
- [ ] `train_p3_centralized.rs` is deleted; no references remain (`grep -rn train_p3_centralized` returns empty).
- [ ] Both flows compile and the existing `cargo test --features training` suite still passes.
- [ ] Smoke run with `--centralized-critic --critic-vf-coef 0.5` produces a finite, positive `centralized_value_loss` in stats output.
- [ ] Smoke run without `--centralized-critic` matches the pre-merge behavior of `train_p3.rs`.
- [ ] Net LOC reduction: target -250 LOC (327 deleted, ~75 added back as CLI flags + branching). Stretch target: -280.

## Test plan

- `LIBTORCH_USE_PYTORCH=1 cargo build --features "training env-bucket-brigade" --example train_p3`
- `LIBTORCH_USE_PYTORCH=1 cargo run --features "training env-bucket-brigade" --example train_p3 -- --scenario default --seed 42 --num-updates 2` — confirm no-critic baseline runs.
- `LIBTORCH_USE_PYTORCH=1 cargo run --features "training env-bucket-brigade" --example train_p3 -- --scenario default --seed 42 --num-updates 2 --centralized-critic` — confirm critic-on path runs and surfaces `centralized_value_loss`.

## Out of scope

- Renaming the binary. Keep it as `train_p3`.
- Adding new training functionality. Pure dedup.
- The `centralized_critic` library code under `src/multi_agent/centralized_critic.rs` — leave it alone.

## Source

Judge review of PR #71: https://github.com/rjwalters/thrust/pull/71#issuecomment-4651918740 (the "Other observations" section).

### PR DESCRIPTION
**Pull request by rjwalters**
## Summary

Collapses `examples/games/bucket_brigade/train_p3_centralized.rs` into `train_p3.rs` behind a `--centralized-critic` CLI flag, per the recommended approach in the issue. The shared core (env construction, redundancy-penalty aux loss, training loop, checkpoint I/O) lives in one place; the two modes diverge only at trainer construction (`JointMultiAgentTrainer::new` vs `::new_with_centralized_critic`).

- Added `--centralized-critic`, `--critic-hidden-dim`, `--critic-lr`, `--critic-vf-coef`, `--critic-clip-range-vf` to `train_p3.rs`'s arg parser. Also accepts `--num-updates` as an alias for `--num-iterations` (matches the smoke-test flag in the issue's test plan).
- Deleted `examples/games/bucket_brigade/train_p3_centralized.rs`.
- Removed the `[[example]] train_p3_centralized` entry from `Cargo.toml` and replaced it with the unified `train_p3` entry (active, mirroring the previous active stance of `train_p3_centralized`).
- Updated the module docs to describe both modes.
- `grep -rn train_p3_centralized` returns empty.

Net delta: **-234 LOC** (122 insertions, 356 deletions). Just under the -250 target but within striking distance; the remainder is unavoidable scaffolding (the new flag-parsing branch, the conditional trainer construction, and the per-iter critic-stats line).

## Smoke tests (from the issue's test plan)

Both smoke runs succeed. Because the `env-bucket-brigade` feature is gated off in the published manifest (see `Cargo.toml` notes on `bucket-brigade-core` being a path-only dep that blocks `cargo publish`), I temporarily uncommented the feature locally to run these and reverted before committing. The committed `Cargo.toml` keeps `bucket-brigade-core` and `env-bucket-brigade = [...]` commented out; `cargo publish --dry-run` confirmed clean afterward.

No-critic:
```
== thrust P3: scenario=default lambda_red=0.01 seed=42 num_agents=4 ==
obs_dim=46 action_dims=[10, 2, 2] device=Cpu
  iter    0 | team_reward -409.860 | red_loss 0.0755 | total 7855791.000 | 0.4s
  iter    1 | team_reward -387.321 | red_loss 0.0726 | total 7487053.000 | 0.4s
```

Centralized critic (note `c_v_loss` positive and finite, per acceptance criterion):
```
== thrust P3 (centralized critic): scenario=default lambda_red=0.01 seed=42 num_agents=4 ==
obs_dim=46 action_dims=[10, 2, 2] device=Cpu
centralized critic: hidden_dim=64 lr=0.0003 vf_coef=0.5 clip_range_vf=0
  iter    0 | team_reward -439.739 | red_loss 0.0987 | c_v_loss 4264053.7500 | c_ev +0.000 | total 10991706.000 | 0.4s
  iter    1 | team_reward -420.924 | red_loss 0.0906 | c_v_loss 4028697.5625 | c_ev +0.000 | total 10295942.906 | 0.4s
```

`cargo test --features training`: 216 lib unit tests + integration suites all pass, 0 failures.

## Test plan

- [x] `cargo build --features "training env-bucket-brigade" --example train_p3` succeeds.
- [x] No-critic smoke run completes 2 updates and saves checkpoints.
- [x] `--centralized-critic` smoke run completes 2 updates, surfaces positive finite `centralized_value_loss`, and saves both per-agent and critic checkpoints.
- [x] `cargo test --features training` passes.
- [x] `cargo publish --dry-run` still succeeds (feature gates preserved).
- [x] `grep -rn train_p3_centralized` returns empty.

Closes #88

Co-Authored-By: Claude Opus 4.7 (1M context) <noreply@anthropic.com>

### CODE REVIEW

**Comment by rjwalters** [comment]
Approved.

**Verdict**: Mechanical dedup, no behavior drift. Verified:

- `examples/games/bucket_brigade/train_p3_centralized.rs` deleted from origin/feature/issue-88; `grep -rn train_p3_centralized` returns empty across the worktree.
- Unified `train_p3.rs` adds `--centralized-critic`, `--critic-hidden-dim`, `--critic-lr`, `--critic-vf-coef`, `--critic-clip-range-vf`, plus the `--num-updates` alias for `--num-iterations`. Pre-existing `--num-iterations` parse path is preserved (line 144), so existing scripts are unaffected.
- Trainer construction branches correctly at line 298: critic-on builds `CentralizedCritic::new_on_device` + optimizer and calls `JointMultiAgentTrainer::new_with_centralized_critic(policies, optimizers, critic, critic_opt, trainer_config)`; critic-off calls `::new(policies, optimizers, trainer_config)`. `JointTrainerConfig.centralized_critic` is set to `Some(critic_cfg.clone())` only in the toggled branch (line 291). All other paths (env construction, `BbJointEnv` shim, redundancy-penalty aux loss, rollout/update loop, per-iter logging, checkpoint I/O for both per-agent and critic) are shared.
- Module docs cover both modes; output-dir auto-redirects to `runs/p3_centralized` when the flag is set (lines 130-137) to prevent clobbering — nice touch.
- **`env-bucket-brigade` gating intact**: `Cargo.toml` line 120 (bucket-brigade-core dep) and line 145 (feature definition) both remain commented out. The active `[[example]]` block at lines 287-291 is `train_p3` (replacing the previously-active `train_p3_centralized` entry — equivalent active stance, just renamed), `required-features = ["training", "env-bucket-brigade"]`. `cargo publish --dry-run --no-default-features --allow-dirty` succeeds from the worktree.
- All 7 CI checks green (Check x2, Rustfmt, Documentation, Security Audit, Tests Linux+libtorch, Tests macOS+libtorch). `mergeStateStatus=CLEAN`, `mergeable=MERGEABLE`.

Net -226 LOC; the dedup is faithful and the publish-clean stance is preserved.
